src/librustc_session/lint/builtin.rs

```diff
@@ -475,7 +475,7 @@ declare_lint! {
 declare_lint! {
     pub INDIRECT_STRUCTURAL_MATCH,
     Warn,
-    "constant used in pattern contains value of non-structural-match type",
```

Comment:
Oh so it's not just references?
Also shouldn't the lint name be about what is wrong with the code? So something like `INDIRECT_NOT_STRUCTURAL_MATCH`?

Reply:
the lint is pre-existing, renaming it doesn't seem worth any confusion here. It's just a future incompat lint, not something users should actually interact with.

Reply:
Oh I see, somehow I thought this is a new lint.